File: src/LinkedList/SimpleLinkedList/SimpleLinkedList.java
package LinkedList.SimpleLinkedList;

public class SimpleLinkedList<T> implements LinkedListInterface<T>{

	private Node<T> head;
	
	public SimpleLinkedList() {
		this.head =  new Node<T>();
	}
	
	@Override
	public boolean isEmpty() {
		return this.head.isNIL();
	}

	@Override
	public int size() {
		int size = 0;
		Node<T> aux = this.head;
		while(!aux.isNIL()){
			aux = aux.getNext();
			size++;
		}
		return size;
	}

	@Override
	public T search(T element) {
		Node<T> aux = this.head;
		while(!aux.isNIL()){
			if(aux.getData().equals(element)){
				return aux.getData();
			}
			aux = aux.getNext();
		}
		return null;
	}

	@Override
	public void insert(T element) {
		if(element != null){
			Node<T> aux = this.head;
			while(!aux.isNIL()){
				aux = aux.getNext();
			}
			aux.setData(element);
			aux.setNext(new Node<T>());
		}
	}

	@Override
	public void remove(T element) {
		if(element != null){
			if(element.equals(this.head.getData())){
				this.head = this.head.getNext();
			}else{
				Node<T> aux = this.head;
				while(!aux.isNIL()){
					if(aux.getData().equals(element)){
						aux.setData(aux.getNext().getData());
						aux.setNext(aux.getNext().getNext());
						break;
					}
					aux = aux.getNext();
				}
			}
		}
	}

	@Override
	public T[] toArray() {
		T[] array = (T[]) new Object[this.size()];
		Node<T> aux = this.head;
		int i = 0;
		while(!aux.isNIL()){
			array[i++] = aux.getData();
			aux = aux.getNext();
		}
		return array;
	}
}

File: Python/LinkedList/SimpleLinkedList/SimpleLinkedList.py
from Node import Node

class SimpleLinkedList():
	
	def __init__(self):
		self.__head = Node()
	
	
	def isEmpty(self):
		return self.__head.isEmpty()
	

	def size(self):
		size = 0
		aux = self.__head
		while(not(self.__head.isEmpty())):
			aux = aux.getNext()
			size = size + 1
		return size

	def search(self,element):
		aux = self.__head
		while(not(aux.isEmpty())):
			if(aux.getData() == element):
				return aux.getData()
			aux = aux.getNext()
		return None

	
	def insert(self,element):
		if(element != None):
			aux = self.__head
			while(not(aux.isEmpty())):
				aux = aux.getNext()
			aux.setData(element)
			aux.setNext(Node())
	
	def remove(self,element):
		if(element != None):
			if(element == self.__head.getData()):
				self.__head = self.__head.getNext()
			else:
				aux = self.__head
				while(not(aux.isEmpty())):
					if(aux.getData() == element):
						aux.setData(aux.getNext().getData())
						aux.setNext(aux.getNext().getNext())
						break;
					aux = aux.getNext()

	
	def toArray(self):
		array = []
		aux = self.__head
		while(not(aux.isEmpty())):
			array.append(aux.getData())
			aux = aux.getNext()
		return array
	
